### PR TITLE
Dm 2702

### PR DESCRIPTION
**Pull request by JoshingYou1**
### JIRA issue link
https://agile6.atlassian.net/browse/DM-2702

## Description - what does this code do?
Refactors code to utilize the `Visn` and `VaFacility` DB tables as opposed to the `vamc.json` file.

## Testing done - how did you test it/steps on how can another person can test it 
1. Run `rails db:migrate`.
2. Run `rails diffusion_history:assign_va_facility_to_diffusion_histories`.
3. Run `rails practice_origin_facilities:assign_va_facility_to_practice_origin_facilities`.
4. Go to all of the pages where we use VISN and VA facility data (search, explore, home map, practice map, PE introduction, PE adoptions, etc.) and confirm that everything still functions as intended.

## Acceptance criteria
- Not using JSON files
- Using VISN and VA Facilities DB Models for source of truth
- Fix up tests
- Link facility model with Diffusion history

## Definition of done
- [X] Unit tests written (if applicable)
- [X] e2e/accessibility tests written (if applicable)
- [X] Events are logged appropriately
- [X] Documentation has been updated, if applicable
- [X] A link has been provided to the originating JIRA issue
- [X] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

**Review by emyl3 — COMMENTED** [review]
just a couple of small things! GREAT WORK on this Josh. Thank you so much for cleaning this up 🧹

File: app/models/user.rb [hunk 1]
```diff
@@ -165,25 +165,25 @@ def self.authenticate_ldap(domain_username)
         user.phone_number = entry[:telephoneNumber][0]
 
         # attempt to resolve where the User is VA facility wise
-        facilities = JSON.parse(File.read("#{Rails.root}/lib/assets/vamc.json"))
+        facilities = VaFacility.cached_va_facilities
```
[the remark this answers is not in the document]
Comment (JoshingYou1) [reply app/models/user.rb:168]: Good point! Updated!

File: app/models/va_facility.rb [hunk 2]
```diff
@@ -10,16 +11,21 @@ class VaFacility < ApplicationRecord
   scope :get_by_visn, -> (visn) { cached_va_facilities.where(visn: visn) }
   scope :get_classification_counts, -> (facility_type) { where(classification: facility_type).size }
   scope :get_classifications, -> { pluck(:classification).uniq }
-  scope :get_station_numbers, -> { pluck(:station_number) }
+  scope :get_ids, -> { pluck(:id) }
   scope :get_locations, -> { order(:street_address_state).pluck(:street_address_state).uniq }
   scope :get_complexity, -> { order(:fy17_parent_station_complexity_level).pluck(:fy17_parent_station_complexity_level).uniq }
+  scope :get_relevant_attributes, -> {
```
[the remark this answers is not in the document]
Comment (JoshingYou1) [reply app/models/va_facility.rb:17]: 😄

File: app/views/practices/_search.js.erb [hunk 3]
```diff
@@ -334,24 +349,25 @@ function searchPracticesPage() {
             var stationNumbers = [];
             var adoptionFacilityArray = [];
             if (adoptingFacility.includes('VISN')) {
+                var adoptingFacilityVisn = getVisn(visns, adoptingFacility);
+
+                console.log('adoptingFacilityVisn', adoptingFacilityVisn)
```
[the remark this answers is not in the document]
Comment (JoshingYou1) [reply app/views/practices/_search.js.erb:354]: These have been removed!

File: spec/features/search_spec.rb [hunk 4]
```diff
@@ -1,9 +1,37 @@
 require 'rails_helper'
+require 'rake'
 
 describe 'Search', type: :feature do
   before do
-    Rake::Task['visns:create_visns_and_transfer_data'].execute
```
[the remark this answers is not in the document]
Comment (JoshingYou1) [reply spec/features/search_spec.rb:5]: 😄

File: app/models/user.rb (hunk 1 above)
[the remark this answers is not in the document]
Comment (emyl3) [reply app/models/user.rb]: Thank you!!

File: app/views/practices/_search.js.erb (hunk 3 above)
[the remark this answers is not in the document]
Comment (emyl3) [reply app/views/practices/_search.js.erb]: Thank you!!

**Review by emyl3 — APPROVED** [review]
YAY for cleaning this up!!! 🧹 👍

**Review by bradjohnson92008 — APPROVED** [review]
VISNS and Facilities working as expected - Great job Josh.

**Pull request merged**